grammar.ebnf

```diff
@@ -33,14 +42,54 @@ Statement = "{", {Statement}, "}"
   because we can achieve that as:
   Arr arr = a.b;  // arr is a pointer under the hood - any change to its
   elements will be reflected to a.b arr = a.b;
-*/
+*)
+
 LValue = Identifier, [ ".", Identifier ];
 
+Expression = FunctionCall
+           | NotExpression
+           | ArrayLookupExpression
+           | BinaryExpression;
+
+FunctionCall = Identifier, "(", FuncCallArgList, ")";
+
 FuncCallArgList = [ Expression, {",", Expression} ];
 
+(*
+  Expression precedence is simulated by using levels.
+  The higher the level, the bigger the precedence.
+*)
+
+BinaryExpression = Level1Expr, [ "&&", Level1Expr, [BinaryExpression]];
+
+Level1Expr = Level2Expr, [ "||", Level2Expr, [Level1Expr]];
+
+Level2Expr = Level3Expr, [ ( "<" || "<=" || ">" || ">=" || "==" || "!="), Level3Expr, [Level2Expr] ];
+
+Level3Expr = Level4Expr, [ ("+" || "-"), Level4Expr, [Level3Expr]];
+
+Level4Expr = ExpressionTerm, [ ("*" || "/" || "%"), ExpressionTerm, [Level4Expr]];
+
+
+NotExpression = { "!" }, ExpressionTerm;
+
+ArrayLookupExpression = ExpressionTerm, "[", ExpressionTerm, "]";
```

Comment:
Keep the LHS as just `Identifier` to make it simple but the RHS (i.e. the part inside `[]`) should be `Expression`).